controllers/user.controller.js

```diff
@@ -0,0 +1,61 @@
+const { CastError } = require("mongoose");
+const bcrypt = require("bcryptjs");
+const User = require("../models/user.schema");
+const { postUserSchema, patchUserSchema } = require("../validation/user.validation");
+const logger = require("../config/logger");
+
+// user create
+const createUser = async (req, res) => {
+  try {
+    const { error, value } = postUserSchema.validate(req.body);
+    if (error) {
+      return res.status(400).send({ message: error.message });
+    }
+    value.password = await bcrypt.hash(value.password, 10);
```

Comment:
This should be done in before Save hook of User Model: https://stackoverflow.com/a/14595363